services/storehost/extMgr.go

```diff
@@ -783,39 +776,57 @@ func (ext *extentContext) delete() {
 	ext.store.DeleteExtent()
 }
 
-func (ext *extentContext) readBeginSeqNum() (beginSeqNum int64, err error) {
+func (ext *extentContext) initFirstMsg() error {
 
 	// FIXME: for timer-queues, this would return the seqNum of the first
 	// _available_ message to deliver, and *not* the first enqueued message.
 
 	_, key, err := ext.store.SeekFirst()
 
-	if err == nil && key != storage.InvalidKey {
-		_, beginSeqNum = ext.deconstructKey(key)
+	if err == nil && key != storage.InvalidKey &&
```

Comment:
positive assertions are more readable then negative assertions. In addition,
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

So, prefer

if err !=nil || key == storage.InvalidKey || ext.isSealExtentKey(key) {
     // init
    return
}